packages/data/lib/repository/movie_repository.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:domain/model/data_model.dart';
+import '../service/api_base_service.dart';
+import '../service/service_payload.dart';
+import 'package:domain/repository/movie_repository.dart';
+
+class NetworkRepositoryImpl implements NetworkRepository {
+  final ApiBaseService<ServicePayload> _movieApiService;
+
+  NetworkRepositoryImpl(this._movieApiService);
+
+  @override
+  Future<GetDataResponse> getMovieData(
+      {required String apiPath, int? itemCount}) async {
+    final response = await _movieApiService.get(
+      apiPath,
+      queryParameters: {
+        'extended': 'full',
+        'page': 1,
+        'limit': itemCount,
+      },
```

Comment:
1) u shouldn't create params here. u should do it at usecase/bloc level (if it is possible to do it in usecase - do it in it, else - bloc)
2) seems that u take first 20 (default page size) from the first page twice - while first api call and while second. it seems redundant

Reply:
2. The first api call I use only for get data from headers - I don't save any response. But in second req I request all data and save it to my empty list.  We will do second request anyway, so I don't see any benefits in saving first 10 movies from 1st request and save 40 instead of 50 in second request. Is this affects performance? Maybe for large requests it is. Or it is a best practice to request less info if it's possible?